integration/system/options_test.go

```diff
@@ -0,0 +1,118 @@
+package system // import "github.com/docker/docker/integration/system"
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api"
+	"github.com/docker/docker/api/types"
+	req "github.com/docker/docker/integration-cli/request"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOptionsRoute(t *testing.T) {
+	resp, _, err := req.Do("/", req.Method(http.MethodOptions))
+	require.NoError(t, err)
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+}
+
+func TestGetEnabledCORS(t *testing.T) {
+	resp, body, err := req.Get("/version")
+	require.NoError(t, err)
+	defer body.Close()
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+	// TODO: @runcom incomplete tests, why old integration tests had this headers
+	// and here none of the headers below are in the response?
+	//c.Log(res.Header)
+	//c.Assert(res.Header.Get("Access-Control-Allow-Origin"), check.Equals, "*")
+	//c.Assert(res.Header.Get("Access-Control-Allow-Headers"), check.Equals, "Origin, X-Requested-With, Content-Type, Accept, X-Registry-Auth")
+}
+
+func TestClientVersionOldNotSupported(t *testing.T) {
+	skip.If(t, testEnv.OSType != runtime.GOOS, "Daemon platform doesn't match test platform")
+
+	skip.If(t, api.MinVersion == api.DefaultVersion, "API MinVersion==DefaultVersion")
+
+	v := strings.Split(api.MinVersion, ".")
+	vMinInt, err := strconv.Atoi(v[1])
+	require.NoError(t, err)
+	vMinInt--
+	v[1] = strconv.Itoa(vMinInt)
+	version := strings.Join(v, ".")
+
+	resp, body, err := req.Get("/v" + version + "/version")
+	require.NoError(t, err)
+	defer body.Close()
+	assert.Equal(t, resp.StatusCode, http.StatusBadRequest)
+
+	expected := fmt.Sprintf("client version %s is too old. Minimum supported API version is %s, please upgrade your client to a newer version", version, api.MinVersion)
+	content, err := ioutil.ReadAll(body)
+	require.NoError(t, err)
+	assert.Contains(t, strings.TrimSpace(string(content)), expected)
+}
+
+func TestErrorJSON(t *testing.T) {
+	resp, body, err := req.Post("/containers/create", req.JSONBody(struct{}{}))
+	require.NoError(t, err)
+	defer body.Close()
+	assert.Equal(t, resp.StatusCode, http.StatusBadRequest)
+	assert.Equal(t, resp.Header.Get("Content-Type"), "application/json")
+
+	b, err := req.ReadBody(body)
+	require.NoError(t, err)
+	var r types.ErrorResponse
+	err = json.Unmarshal(b, &r)
+	require.NoError(t, err)
+	assert.Equal(t, strings.TrimSpace(r.Message), "Config cannot be empty in order to create a container")
```

Comment:
Not sure what this is supposed to be testing. Is it testing the container create API? It could be a unit test of the API router.